app/models/classified_listing.rb

```diff
@@ -40,34 +27,38 @@ class ClassifiedListing < ApplicationRecord
   validates :location, length: { maximum: 32 }
   validate :restrict_markdown_input
   validate :validate_tags
-  validate :validate_category
 
   scope :published, -> { where(published: true) }
 
-  def self.cost_by_category(category)
-    categories_available.dig(category, :cost) || 0
-  end
-
-  def author
-    organization || user
+  # TODO: refactor this class method block
```

Comment:
In order to keep the number of moving parts in each PR relatively low I decided to keep the old interface and just swap out the implementation.